src/protocol/sender/mod.rs

```diff
@@ -0,0 +1,499 @@
+use std::collections::VecDeque;
+use std::net::SocketAddr;
+use std::time::{Duration, Instant};
+
+use bytes::Bytes;
+use log::debug;
+use log::{trace, warn};
+
+use crate::loss_compression::decompress_loss_list;
+use crate::packet::{ControlTypes, HandshakeControlInfo, SrtControlPacket};
+use crate::protocol::handshake::Handshake;
+use crate::protocol::sender::buffers::*;
+use crate::protocol::Timer;
+use crate::Packet::*;
+use crate::{
+    CCData, CongestCtrl, ConnectionSettings, ControlPacket, DataPacket, Packet, SeqNumber,
+    SrtCongestCtrl, Stats,
+};
+
+pub mod buffers;
```

Comment:
Pub? Are the buffers part of the public interface?

Reply:
No, they aren't part of the public interface. I believe this was a residual of the process I took to refactoring the code. I did it one small step at the time running all the tests each change, which included extracting the buffer classes and associated code as intermediate steps. I neglected to go back and clean up the public interface of the modules. I'll fix this, but I imagine there will be further improvement to be had besides this. Consider this a first pass at improving the code. I share your desire for a "clean" public interface, but it will evolve over time. As I mentioned before, my intention is to have three high level modules, each with an appropriate public interfaces: a packet (or wire?) module, a protocol (e.g. state machines, algorithms, timers, etc.), and a tokio module (e.g. builders, errors, SrtListener and SrtStream, etc. to mimic the tokio TCP design).

Reply:
Sounds reasonable.